### DISCUSSION TITLE
The migration guide lists "migrating routing hooks" as step 5, but it seems like it should be step 1.

### DISCUSSION THREAD

**Discussion by ethanryan** [vercel/next.js#50403] [App Router]
The migration guide lists "migrating routing hooks" as step 5, but it seems like it should be step 1.

https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration#step-5-migrating-routing-hooks

Migrating all those routing hooks first should make the other steps easier.

Is there any reason NOT to migrate routing hooks first, before the other steps in the app router migration guide?

**Comment by devjiwonchoi** [comment]
In my opinion, it is because the new Routing Hooks are designed to use in the `app` directory and will cause an error if the prerequisites of the previous Steps are not fulfilled. 

For example, the [`useRouter`](https://nextjs.org/docs/pages/api-reference/functions/use-router) hook from `next/router` used in `Page Router`  is overlapped with the [`useRouter`](https://nextjs.org/docs/app/api-reference/functions/use-router) hook in `next/navigation` used in `App Router`. 

If migrating the hooks were on Step 1, it might cause confusion to the people who are not familiar with the `App Router` directory structure.

**Reply by ethanryan** [reply]
But using the `useRouter` hook from `next/navigation` can be used in components rendered by pages in the Page Router, correct?

If they **can't** be, we'll need to have two versions of all our components that use the useRouter hook: one for use in the Page Router, one for use in the App Router.

If the `useRouter` hook from `next/navigation` **can** be used in components rendered by pages in the Page Router: doesn't it make sense to first update all uses of the `useRouter` hook?

**Reply by devjiwonchoi** [reply]
Since the point is to migrate from `Page Router` to `App Router`, the developer should recognize that there are significant differences between the two.

The `useRouter` from `next/navigation` can only be used inside the Client Components, so it requires `'use client';` on the top of the file.

>The useRouter hook allows you to programmatically change routes inside [Client Components](https://nextjs.org/docs/getting-started/react-essentials).

On the other hand, the `useRouter` from `next/router` is for any function component.

>If you want to access the [router object](https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object) inside any function component in your app, you can use the useRouter hook, ...

Also, the [docs](https://nextjs.org/docs/app/api-reference/functions/use-router) states to ensure that in `App Router`, import `useRouter` from `next/navigation` and not `next/router`.

>The new useRouter hook should be imported from next/navigation and not next/router

Therefore in order to reduce confusion for the developer who's migrating from the `Page Router` to `App Router`, it seems to be essential to adopt the `App Router` in a certain order which is guided in the official docs.

**Reply by ethanryan** [reply]
Thanks for your replies @devjiwonchoi. I'm still not convinced there's any reason not to migrate the routing hooks first, before the other steps listed in the App Router migration docs, as I think it will ultimately save us time. Was curious if any other devs came to a similar conclusion, or had a strong reason not to take this approach. But I appreciate your thoughts.
